libs/perun/services/src/lib/store.service.ts

```diff
@@ -62,73 +62,41 @@ export class StoreService {
   }
 
   getProperty<T extends keyof PerunConfig>(key: T): PerunConfig[T] {
+    if (!this.instanceConfig || !this.defaultConfig) {
+      return null;
+    }
+
     const configs: PerunConfig[] = [
       this.instanceConfig?.brandings?.[this.branding],
       this.instanceConfig,
-      this.defaultConfig,
     ];
 
+    const defaultValue: PerunConfig[T] = this.defaultConfig[key];
     let currentValue: PerunConfig[T] = null;
     for (const config of configs) {
       if (config && (currentValue === null || currentValue === undefined)) {
         currentValue = config[key];
       }
     }
 
-    return currentValue;
-  }
-
-  /**
-   * @deprecated - Use method `getProperty` instead.
-   */
-  /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment */
-  get(...keys: string[]): any {
-    let currentValue: string;
-
-    if (this.branding !== '') {
-      const brandingConfig: PerunConfig = this.instanceConfig.brandings[this.branding];
-      for (let i = 0; i < keys.length; ++i) {
-        if (i === 0) {
-          currentValue = brandingConfig[keys[i]];
-        } else {
-          if (currentValue === undefined) {
-            break;
-          }
-          currentValue = currentValue[keys[i]];
-        }
-      }
+    if (!currentValue) {
+      return defaultValue;
     }
 
-    if (this.instanceConfig !== undefined && currentValue === undefined) {
-      for (let i = 0; i < keys.length; ++i) {
-        if (i === 0) {
-          currentValue = this.instanceConfig[keys[i]];
-        } else {
-          if (currentValue === undefined) {
-            break;
-          }
-          currentValue = currentValue[keys[i]];
-        }
-      }
-    }
+    return this.addMissingValuesToProperty(currentValue, defaultValue);
+  }
 
-    if (this.defaultConfig === undefined) {
-      return undefined;
-    }
-    if (currentValue === undefined) {
-      for (let i = 0; i < keys.length; ++i) {
-        if (i === 0) {
-          currentValue = this.defaultConfig[keys[i]];
-        } else {
-          if (currentValue === undefined) {
-            // console.error('Missing value in default config: ' + keys);
-            break;
-          }
-          currentValue = currentValue[keys[i]];
-        }
+  addMissingValuesToProperty<T extends keyof PerunConfig>(
+    value: PerunConfig[T],
+    defaultValue: PerunConfig[T]
+  ): PerunConfig[T] {
+    if (typeof value === 'object' && !Array.isArray(value) && value) {
+      for (const key of Object.keys(defaultValue)) {
+        defaultValue[key] = this.addMissingValuesToProperty(value[key], defaultValue[key]);
```

Comment:
I think you can use something like this to fix eslint errors.
```suggestion
        defaultValue[key] = this.addMissingValuesToProperty(
           value[key] as PerunConfig[T], 
           defaultValue[key] as PerunConfig[T]
        );
```

Reply:
I already said this to @sarkapalkovicova but the issue here is that on consecutive passes the values are no longer item of PerunConfig but some nested item. So the definition of function has to be changed.